### PR TITLE
.gitignore: Remove `package-lock.json`

### LINKED ISSUE #2508: Remove `package-lock.json` from gitignore
npm clearly states that this file should be commited. From `npm help package-lock.json`:

> package-lock.json is automatically generated for any operations where npm modifies either the node_modules tree, or package.json. It describes the exact tree that was generated, such that subsequent installs are able to generate identical trees, regardless of intermediate dependency updates.
>
> This file is intended to be committed into source repositories, and serves various purposes:
>
> - Describe a single representation of a dependency tree such that teammates, deployments, and continuous integration are guaranteed to install exactly the same dependencies.
> - Provide a facility for users to "time-travel" to previous states of node_modules without having to commit the directory itself.
> - To facilitate greater visibility of tree changes through readable source control diffs.
> - And optimize the installation process by allowing npm to skip repeated metadata resolutions for previously-installed packages.
>
> One key detail about package-lock.json is that it cannot be published, and it will be ignored if found in any place other than the toplevel package. It shares a format with npm-shrinkwrap.json(5), which is essentially the same file, but allows publication. This is not recommended unless deploying a CLI tool or otherwise using the publication process for producing production packages.
>
> If both package-lock.json and npm-shrinkwrap.json are present in the root of a package, package-lock.json will be completely ignored.

This issue is about reverting the change in https://github.com/coala/coala-bears/pull/2494 again.

### PR DESCRIPTION
**Pull request by KaplanAlex**
Removed package-lock.json from .gitignore so this file will be committed, as npm clearly states that this file should be commited.

Closes https://github.com/coala/coala-bears/issues/2508

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by KaplanAlex** [comment]
I re-did the commit message, but when I tried to push the commit I was instructed to merge “my-first-pull-request” with “my-first-pull-request” which sounds innocent enough. However, now the merge commit message is what appears instead of my new, correctly formed message. Does anyone know what I should do? Will I need to start a new pull request?

**Comment by KaplanAlex** [comment]
Also, any idea why all the tests are failing? I cloned the libraries and only made the required change to the .gitignore.

**Comment by jsecrest** [comment]
You should never close a pull request. You should just get help fixing this one. I also see that there are a few commits in this branch. When you fixed your commit message you should have amended the commit. After amending the commit then you force push it. (see the [follow up section in the coala git tutorial](https://api.coala.io/en/latest/Developers/Git_Basics.html#follow-up)) As it is maybe you need to squash them? (Not sure, I'm still learning.)

A great place to get help is here https://gitter.im/coala/coala - there are a lot of active contributors who monitor that chat.

**Comment by li-boxuan** [comment]
As @jsecrest points out, you need to squash your commit messages and then do a force push.

**Review by li-boxuan — CHANGES_REQUESTED** [review]
Add and commit `package-lock.json`

**Review by sks444 — CHANGES_REQUESTED** [review]
Remove backtick from `package-lock.json` in your commit shortlog.

And in your commit body, commited --> committed.

**Comment by Makman2** [comment]
ack 036eff66edb5bd49ea6282c8637e8868fec4b69f

**Comment by li-boxuan** [comment]
@Makman2 Shouldn't `package-lock.json` get added and committed?

**Comment by KaplanAlex** [comment]
I can commit my package-lock.josn but I'm worried something is wrong with my build. Travis fails here and 17 tests fail locally when I run pytest. I'm not sure what is causing this issue, as the only difference between my fork and coala-bears is this change. Either way, this change will commit package-lock.json in the future following project builds.

**Review by bekicot — COMMENTED** [review]
You should also run npm install and add package-lock.json into the commit.

@KaplanAlex The build error is not your fault 😄 .
Follow this https://github.com/coala/coala/issues/5449

TLDR
 it needs
https://gitlab.com/coala/mobans/merge_requests/22
to be merged and synced.

**Comment by KaplanAlex** [comment]
Oh okay I see now. Thanks for pointing me to the issue.

**Comment by Makman2** [comment]
@li-boxuan seems we never had a `package-lock.json` committed. I would do that in another issue, that one would also include to verify that `package-lock.json` has the right content^^

**Comment by KaplanAlex** [comment]
So I should remove package_lock.json?

**Comment by li-boxuan** [comment]
@KaplanAlex Yes, please remove that. ;)

**Comment by bekicot** [comment]
@li-boxuan Why removing it?

**Comment by KaplanAlex** [comment]
I'm sorry I still can't tell if I should remove it or not. I'm happy to do whatever

**Comment by jayvdb** [comment]
We need to wait until https://github.com/coala/coala-bears/pull/2530 anyway.

The rough decision on gitter was that we would proceed with this patch, and check in a package-lock.json , and see what headaches it brings, if any.

Note that https://github.com/coala/coala-bears/pull/2531 also removes it.

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot ff
